src/deepagents/interrupt.py

```
@@ -0,0 +1,95 @@
+        interrupt_tool_calls = []
+        auto_approved_tool_calls = []
+        
+        for tool_call in last_message.tool_calls:
+            tool_name = tool_call["name"]
+            if tool_name in tool_configs:
+                interrupt_tool_calls.append(tool_call)
+            else:
+                auto_approved_tool_calls.append(tool_call)
+
+        # If no interrupts needed, return early
+        if not interrupt_tool_calls:
+            return state
+
+        approved_tool_calls = auto_approved_tool_calls.copy()
+
+        # Process all tool calls that need interrupts in parallel
+        requests = []
+        
+        for tool_call in interrupt_tool_calls:
+            tool_name = tool_call["name"]
+            tool_args = tool_call["args"]
+            description = f"{message_prefix}\n\nTool: {tool_name}\nArgs: {tool_args}"
+            tool_config = tool_configs[tool_name]
+
+            request: HumanInterrupt = {
+                "action_request": ActionRequest(
+                    action=tool_name,
+                    args=tool_args,
+                ),
+                "config": tool_config,
+                "description": description,
+            }
+            requests.append(request)
+
+        responses: List[HumanResponse] = interrupt(requests)
+        
+        for i, response in enumerate(responses):
+            tool_call = interrupt_tool_calls[i]
+            
+            if response["type"] == "accept":
+                approved_tool_calls.append(tool_call)
+            elif response["type"] == "edit":
+                edited: ActionRequest = response["args"]
+                new_tool_call = {
+                    "name": tool_call["name"],
+                    "args": edited["args"],
+                    "id": tool_call["id"],
+                }
+                approved_tool_calls.append(new_tool_call)
+            # For "ignore" and other types, we skip adding the tool call
```

what are other types?

i would maybe list out the other types and throw error if unknown type - in the case we add a new type, would catch that and not silently do something we dont expect

---

accept/edit seems to be the only one's an interrupt would allow, found this here: [https://langchain-ai.github.io/langgraph/how-tos/human_in_the_loop/add-human-in-the-loop/#review-tool-calls](https://langchain-ai.github.io/langgraph/how-tos/human_in_the_loop/add-human-in-the-loop/#review-tool-calls)